zipline/pipeline/factors/statistical.py

```diff
@@ -104,31 +118,43 @@ class RollingSpearman(_RollingCorrelation):
     Most users should call Factor.spearmanr rather than directly construct an
     instance of this class.
     """
-    def compute(self, today, assets, out, factor_data, slice_data):
-        slice_data_column = slice_data[:, 0]
-        for i in range(len(out)):
-            # spearmanr returns the R-value and the P-value.
-            out[i] = spearmanr(factor_data[:, i], slice_data_column)[0]
+    def compute(self, today, assets, out, base_data, target_data):
+        if target_data.shape[1] > 1:
```

Comment:
Cleaner here might be:

```
target_date = np.broadcast_to(target_date, base_data.shape)
for i in range(len(out)):
    ...
```

The same pattern could be used pretty much everywhere in this file.